Index: AvlTree.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>//\r\n// Created by 97254 on 21/11/2022.\r\n//\r\n\r\n#ifndef WET1_AVLTREE_H\r\n#define WET1_AVLTREE_H\r\n\r\n#include <iostream>\r\n\r\ntemplate<class T, class K>\r\nclass AvlTree {\r\n\r\nprivate:\r\n\r\n    struct node {\r\n        T data;\r\n        K key;\r\n        struct node *right;\r\n        struct node *left;\r\n    };\r\n\r\n    node* root;\r\n    void deleteTree(node* r);\r\n\r\n\r\npublic:\r\n    AvlTree(){\r\n        this->root = NULL;\r\n    }\r\n    ~AvlTree();\r\n    node* insert(node* r, T value, K key);\r\n    int height(node* r);\r\n    int bf(node* r);\r\n    node* balance(node* r);\r\n\r\n\r\n\r\n\r\n    node * llRotation( node *parent);\r\n    node * rrRotation( node *parent);\r\n    node * rlRotation( node *parent);\r\n    node * lrRotation( node *parent);\r\n\r\n    void delete( node *p,T data);\r\n    T& find( node *p,T data); //or node*\r\n\r\n#endif //WET1_AVLTREE_H\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/AvlTree.h b/AvlTree.h
--- a/AvlTree.h	(revision bd391b85840ad23f59138df46857361ee7aed666)
+++ b/AvlTree.h	(date 1669474048108)
@@ -32,10 +32,6 @@
     int height(node* r);
     int bf(node* r);
     node* balance(node* r);
-
-
-
-
     node * llRotation( node *parent);
     node * rrRotation( node *parent);
     node * rlRotation( node *parent);
